### PR TITLE
fix(storage): fix metrics compact task size

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

as title

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by Li0k** [comment]
PTAL @Little-Wallace @zwang28

File: src/storage/hummock_sdk/src/compact.rs [hunk 1]
```diff
@@ -101,27 +101,36 @@ pub fn append_sstable_info_to_string(s: &mut String, sstable_info: &SstableInfo)
     }
 }
 
-pub fn estimate_state_for_compaction(task: &CompactTask) -> (u64, usize, u64) {
-    let mut total_memory_size = 0;
-    let mut total_file_count = 0;
+pub fn estimate_state_for_compaction(task: &CompactTask) -> CompactTaskEstimatedState {
```
Comment (zwang28) [inline src/storage/hummock_sdk/src/compact.rs:104]: The function naming `estimate` no longer makes sense.

File: src/storage/src/hummock/compactor/compaction_utils.rs [hunk 2]
```diff
@@ -228,34 +228,20 @@ pub async fn generate_splits(
...
-        .map(|table| table.uncompressed_file_size)
-        .sum::<u64>();
 
     let capacity = std::cmp::min(task.target_file_size as usize, max_target_file_size);
     let total_file_size = (total_input_file_size as f64 * 1.4).round() as usize;
 
-    let c = match task.compression_algorithm {
+    match task.compression_algorithm {
         0 => std::cmp::min(capacity, total_file_size),
         _ => capacity,
```
Comment (zwang28) [inline src/storage/src/hummock/compactor/compaction_utils.rs:245]: Why _ => doesn't need `min`?

File: src/storage/src/hummock/compactor/mod.rs [hunk 3]
```diff
@@ -295,16 +294,13 @@ impl Compactor {
         ) * compact_task.splits.len() as u64;
 
         tracing::info!(
-                "Ready to handle compaction task: {} need memory: {} input_file_counts {} input_file_size {} input_file_size_uncompressed {} total_key_count {} target_level {} compression_algorithm {:?} parallelism {}",
+            "Ready to handle compaction task: {} compact_task_estimated_state {:?} target_level {} compression_algorithm {:?} parallelism {} task_memory_capacity_with_parallelism {}",
```
Comment (zwang28) [inline src/storage/src/hummock/compactor/mod.rs:297]: May implement Display for compact_task_estimated_state.

**Review by zwang28 — APPROVED** [review]
LGTM

File: src/storage/src/hummock/compactor/compaction_utils.rs (hunk 2 above)
Comment (Li0k) [reply src/storage/src/hummock/compactor/compaction_utils.rs:245]: good catch 

Even now, after we only use streaming_uploader, this estimate is not so important.

I'd still like to discuss whether we should use total_input_uncompressed_file_size for estimation? Here _=> the reason for not using min, is that in most cases, the uncompressed_file_size > capacity (compression_algorithm > 0)

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Pull request merged**